_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -1041,11 +1043,76 @@ private function _process_onboarding( $data ) {
 			}
 		}
 
+		// Add/update business address widget
+		if ( isset( $data['businessAddress'] ) ) {
+			//grab the first sidebar and proceed if it is present
+			$first_sidebar = self::get_first_sidebar();
+
+			if( $first_sidebar ) {
+				$title = wp_unslash( $data[ 'businessAddress' ][ 'name' ] );
+				$address = wp_unslash(
+					$data[ 'businessAddress' ][ 'city' ] . ' ' .
+					$data[ 'businessAddress' ][ 'state' ] . ' ' .
+					$data[ 'businessAddress' ][ 'street' ] . ' ' .
+					$data[ 'businessAddress' ][ 'zip' ]
+				);
+				$widget_options = array(
+					'title'   => $title,
+					'address' => $address,
+					'phone'   => '',
+					'hours'   => '',
+					'showmap' => false,
+					'email' => '',
+					'apikey' => '',
+				);
+				$position = '0';
+				$id_base = 'widget_contact_info';
+
+				$has_ba_widget = self::has_business_info_widget( $first_sidebar );
+				if ( $has_ba_widget ) {
+					self::update_widget( $id_base, $first_sidebar, $position, $widget_options);
+				}	else {
+					Jetpack_Widgets::activate_widget( $id_base, $first_sidebar, $position, $widget_options);
+				}
+			}
+		}
+
 		return empty( $error )
 			? ''
 			: join( ', ', $error );
 	}
 
+		public function get_first_sidebar() {
+			$active_sidebars = Jetpack_Widgets::get_active_sidebars();
+
+			unset( $active_sidebars[ 'wp_inactive_widgets' ], $active_sidebars[ 'array_version' ] );
+
+			if ( empty( $active_sidebars ) ) {
+				return false;
+			}
+			return array_shift( array_keys( $active_sidebars ) );
+		}
+
+		public function has_business_info_widget( $sidebar ) {
+			$sidebars_widgets = get_option( 'sidebars_widgets', array() );
+			if ( ! isset( $sidebars_widgets[ $sidebar ] ) ) {
+				return false;
+			}
+
+			foreach ( $sidebars_widgets[ $sidebar ] as $widget ) {
+				if ( strpos( $widget, 'widget_contact_info' ) !== false ) {
+					return true;
+				}
+			}
+			return false;
+		}
+
+		public function update_widget( $id_base, $sidebar, $position, $widget_options ) {
```

Comment:
Should this method be in `_inc/lib/widgets.php` instead? Guess we should figure another name for it? `update_widget_by_id_base` maybe?

Reply:
I put it in a separate method when at the beginning of getting it up and running the code was much longer. Looking at it now, I reckon it is not needed in fact. Slightly re-written it becomes a two-liner and can be used in-place. Do you agree?

Reply:
Not sure what exactly you mean - could you please elaborate?

Reply:
I removed that function and included: https://github.com/Automattic/jetpack/pull/8426/files#diff-2ccec9069213650359171727de9a2471R1073.